packages/react-components/react-utilities/src/hooks/useId.ts

```diff
@@ -37,7 +37,9 @@ export function useId(prefix: string = 'fui-', providedId?: string): string {
   const _useId: () => string | undefined = (React as never)['use' + 'Id'];
 
   if (_useId) {
-    return providedId || `${idPrefix}${prefix}${_useId()}`;
+    const generatedId = _useId()?.replace(/:/g, '');
```

Comment:
nit: The replace can happen conditionally, to not replace even when not needed.

Reply:
Not sure I understand what you mean. If it's that sometimes replace can be executed when there's no colons, that's not the case. From the code that makes the id:
```
function makeId(responseState, treeId, localId) {
    var idPrefix = responseState.idPrefix;
    var id = ':' + idPrefix + 'R' + treeId; // Unless this is the first id at this level, append a number at the end
    // that represents the position of this useId hook among all the useId
    // hooks for this fiber.

    if (localId > 0) {
      id += 'H' + localId.toString(32);
    }

    return id + ':';
  }
```
There will always be a colon. Also the `?` is needed since we don't have the types for use id. If you prefer I could type cast it, but I feel it's not the right way to go.